cdc_replication_test.py

```diff
@@ -0,0 +1,193 @@
+            
+            java -cp replicator.jar com.scylladb.scylla.cdc.replicator.Main -k ks -t tb -s {} -d {} -cl {} 2>&1 | tee replicatorlog
+            EOF
+        """.format(master_node.external_address, replica_node.external_address, 'one'))
+
+        print('Replicator script:\n', replicator_script)
+
+        print('Starting replicator.')
+        res = loader_node.remoter.run(cmd=replicator_script)
+        if res.exit_status != 0:
+            self.fail('Could not start CDC replicator.')
+
+        print("Let C-S run for a while...")
+        time.sleep(30)
+
+        print("Stopping nemesis before bootstrapping a new node.")
+        self.db_cluster.stop_nemesis(timeout=60)
+
+        print("Bootstrapping a new node...")
+        new_node = self.db_cluster.add_nodes(count=1, enable_auto_bootstrap=True)[0]
+        print("Waiting for new node to finish initializing...")
+        self.db_cluster.wait_for_init(node_list=[new_node])
+
+        print("Bootstrapped, restarting nemesis.")
+        self.db_cluster.start_nemesis()
+
+        print("Waiting for C-S to finish...")
+        stress_results = stress_thread.get_results()
+
+        print("Stress results:", [r for r in stress_results])
+
+        print('Waiting for replicator to finish (sleeping 60s)...')
+        time.sleep(60)
+
+        print("Stopping nemesis.")
+        self.db_cluster.stop_nemesis(timeout=60)
+
+        print('Fetching replicator logs.')
+        replicator_log_path = os.path.join(self.logdir, 'replicator.log')
+        loader_node.remoter.receive_files(src='replicatorlog', dst=replicator_log_path)
+
+        print('Comparing table contents using scylla-migrate...')
+        res = loader_node.remoter.run(cmd=
+            './scylla-migrate check --master-address {} --replica-address {}'
+            ' --ignore-schema-difference ks.tb 2>&1 | tee migratelog'.format(
+                    master_node.external_address, replica_node.external_address))
+
+        migrate_log_path = os.path.join(self.logdir, 'scylla-migrate.log')
+        loader_node.remoter.receive_files(src='migratelog', dst=migrate_log_path)
+        with open(migrate_log_path) as f:
+            consistency_ok = 'Consistency check OK.\n' in (line for line in f)
+
+        if not consistency_ok:
+            print('Inconsistency detected.')
+        if res.exit_status != 0:
+            print('scylla-migrate command returned status', res.exit_status)
+
+        if consistency_ok and res.exit_status == 0:
+            print('Consistency check successful.')
+            return
+
+        # Collect more data for analysis
+
+        print('scylla-migrate log:')
+        print_file_to_stdout(migrate_log_path)
+        print('Replicator log:')
+        print_file_to_stdout(replicator_log_path)
+
+        with self.cql_connection_patient(node=master_node) as sess:
+            print('Fetching master table...')
+            res = sess.execute(SimpleStatement('select * from ks.tb',
```

Comment:
Does it work without timing out?

Reply:
It uses paging to perform the query. If I understand correctly, multiple round-trips to the cluster are performed underneath.

A bunch of local runs from my laptop didn't timeout here.